File: solutions/ruby/hamming/1/hamming.rb
# frozen_string_literal: true

# Calculate the Hamming distance between DNA strands
module Hamming
  def self.compute(strand1, strand2)
    raise ArgumentError unless strand1.length == strand2.length

    strand1.chars.zip(strand2.chars).count { |x, y| x != y }
    # strand1.size.times.count { |i| strand1[i] != strand2[i] }
  end
end

File: solutions/python/hamming/1/hamming.py
def distance(strand_a: str, strand_b: str) -> int:
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")
    return sum(a != b for a, b in zip(list(strand_a), list(strand_b), strict=True))
